x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/create_step/progress_stats.tsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, FC, useState, useEffect } from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiProgress, EuiSpacer, EuiText } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useMlKibana } from '../../../../../contexts/kibana';
+import { getDataFrameAnalyticsProgressPhase } from '../../../analytics_management/components/analytics_list/common';
+import { isGetDataFrameAnalyticsStatsResponseOk } from '../../../analytics_management/services/analytics_service/get_analytics';
+import { ml } from '../../../../../services/ml_api_service';
+import { DataFrameAnalyticsId } from '../../../../common/analytics';
+
+export const PROGRESS_REFRESH_INTERVAL_MS = 2000;
+
+export const ProgressStats: FC<{ jobId: DataFrameAnalyticsId }> = ({ jobId }) => {
+  const [initialized, setInitialized] = useState<boolean>(false);
+  const [currentProgress, setCurrentProgress] = useState<{
+    currentPhase: number;
+    progress: number;
+    totalPhases: number;
+  }>(undefined);
+
+  const {
+    services: { notifications },
+  } = useMlKibana();
+
+  const getCurrentStats = () => {
+    function startProgressBar() {
+      const interval = setInterval(async () => {
```

Comment:
If the user navigates away from this page while this interval still exists, it may cause errors to be thrown.
Storing the interval in state which is cleared on component destruction might be safer.
Also, just for extra security against run away intervals, it might be safer to use a `setTimeout` which recreates itself on every call.

Reply:
Good point - switched to `setTimeout` in 9bc31924120d1749680a4d2a25e64a617408774b

Reply:
Ended up keeping the interval so it could be cleared on unmount in the useEffect hook 799856b24d75856a5593c2e330fa8174e4e0fdec